### ISSUE TITLE
One Template Sheet to Multiple Output Sheets

### ISSUE THREAD

**Issue by jebberjeb** [tomfaulhaber/excel-templates#4]
I'd like to use a single template sheet to render multiple sets of data, yielding multiple sheets of different names. 

First question, how would the input data look for this?

    {"Squares" {3 [[1 1] [2 4] [3 9]]}} 

might become

    [{:template-sheet-name "Squares" :output-sheet-name "Foo Squares"} {3 [[1 1] [2 4] [3 9]]}
     {:template-sheet-name "Squares" :output-sheet-name "Bar Squares"} {3 [[5 25] [6 36]]}]

or, maybe add a level of depth to the inner map

    {"Squares" {"Foo Squares" {3 [[1 1] [2 4] [3 9]]}}
                "Bar Squares" {3 [[5 25] [6 36]]}}}

maybe a very verbose data structure more like the former + a macro to expand something friendlier like the current format into the verbose version?

**Comment by tomfaulhaber** [comment]
I was thinking about this after we chatted today.

What I came up with was something like your second, but use a vector instead of a map:

```
{"Squares" [{:sheet-name "Foo Squares" 
             3 [[1 1] [2 4] [3 9]]}
            {:sheet-name "Bar Squares" 
             3 [[5 25] [6 36]]}]}
```

This has the advantage of preserving order and not changing the structure too much.

If there's no `:sheet-name` key:

```
{"Squares" [{3 [[1 1] [2 4] [3 9]]}
            {3 [[5 25] [6 36]]}]}
```

You'll get the names "Squares" and "Squares-1" as names. Ensuing sheets would be "Squares-2" etc.

The code that reads the replacements can then simply convert: 

```
{"Squares" {3 [[1 1] [2 4] [3 9]]}}
```

to 

```
{"Squares" [{3 [[1 1] [2 4] [3 9]]}]}
```

(i.e., wrapping any map values in vectors) and everything is consistent.

One thing I think I'm going to do rsn is write up a real grammar that shows how these variations work.
